backend/src/org/commcare/suite/model/Entry.java

```diff
@@ -41,20 +37,25 @@
      * Serialization only!
      */
     public Entry() {
-
     }
 
-    public Entry(String commandId, DisplayUnit display, Vector<SessionDatum> data, String formNamespace, Hashtable<String, DataInstance> instances,
-                 Vector<StackOperation> stackOperations, AssertionSet assertions) {
+    public Entry(String commandId, DisplayUnit display,
+                 Vector<SessionDatum> data,
+                 Hashtable<String, DataInstance> instances,
+                 Vector<StackOperation> stackOperations,
+                 AssertionSet assertions) {
         this.commandId = commandId == null ? "" : commandId;
         this.display = display;
         this.data = data;
-        xFormNamespace = formNamespace;
         this.instances = instances;
         this.stackOperations = stackOperations;
         this.assertions = assertions;
     }
 
+    public boolean isView() {
+        return this instanceof ViewEntry;
```

Comment:
Wouldn't it make more sense to do this through the polymorphism of the objects? Seems convoluted to couple at this level.